File: movie_back/app/Models/favorite_model.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class favorite_model extends Model
{
    use HasFactory;
    protected $table='favorite';
    protected $fillable=[
        'user_id',
        'movie_id',
    ];
    public function movie(){
        return $this->belongsTo(movie_model::class);
    }
    public function user(){
        return $this->belongsTo(related: User::class);
    }
}

File: movie/lib/model/favorite_model.dart
import 'package:movie/model/movie_model.dart';

class favorite_model {
  final int id;
  final int user_id;
  final int movie_id;
  movie_model movie;
  favorite_model({
    required this.id,
    required this.user_id,
    required this.movie_id,
    required this.movie,
  });
  factory favorite_model.fromjson(Map<String,dynamic>json){
    return favorite_model(
      id:json['id'],
      user_id:json['user_id'],
      movie_id:json['movie_id'],
      movie:movie_model.fromjson(json['movie']),
    );
  }
}
